### ISSUE TITLE
policy_ent_coeff does not seem to work in tf/NPO

### ISSUE THREAD

**Issue by ryanjulian** [rlworkgroup/garage#281]
I can choose any value (or GaussianMLPPolicy params) I want, and it doesn't change the policy standard deviation. e.g. -1.0 won't cause the policy stddev to fall, and 1.0 won't cause it to rise. This may be because a gradient path is broken between the reward function and the policy std network.

**Comment by ryanjulian** [comment]
Things to check:

* Assuming your augmented rewards equation looks like `r + (policy_ent_coeff * policy_entropy)`, ensure that `policy_entropy` is differentiable (symbolic) all the way back to the `std_network` of the policy. This means that your loss function retrieves policy_entropy by purely by feeding actions and observations, not by using the recorded mean/std (e.g. agent_infos, dist_infos).
* Make sure that dist_infos are only used for forward pass calculations (e.g. KL divergence)

**Comment by ryanjulian** [comment]
@CatherineSue were you able to reproduce this?

**Comment by CatherineSue** [comment]
Y

**Comment by ryanjulian** [comment]
For this issue, let's use
* MaxEnt (e.g. augment the reward with an entropy term and a coefficient): \hat r = r + \alpha_1 * H
* log-likelihood estimator: H ~ -logli(action| policy(state))
* Turn off the gradient: H = tf.stop_gradient(H)
